### PR TITLE
changed location of options and views

### LINKED ISSUE #113: Project details should be available across the service (anytime the project is shown, lemme change "unconfirmed", archive, etc)
image

Right now, this is only available at the project roll-up.
Gimme at my staffplan + Project.

### LINKED ISSUE #161: Delete behavior: for projects, for people
If I add a project, anywhere in StaffPlan, and it has a unique name, it should immediately appear in the screen I'm on (or if I'm in the All "People" screen, the screen will redraw and load that new project.

If I add a project, anywhere in StaffPlan, and it has a record already (the same name, the same client), I should have the option to either
1) add it to the view I'm in (I'm restoring it if had been archived/deleted)
or
2) not load the previously created project (show that it already exists) but to rename my new project.

If I delete a project (currently the archive button) from an individual's StaffPlan, then the project goes away just from that person's StaffPlan. It is not deleted from anyone else's. I can add it back by adding the same named project again to that StaffPlan.

If I delete a project from the main Project page, the project is archived across everyone's staffplan (but again, the data isn't deleted, it's just that the Project is not shown anymore - its in archive status).

MORE HERE

### PR DESCRIPTION
**Pull request by HankC138**
changed the location of archive and delete functions and made them accessible on all three main views. And changed the look of the archive button on edit project form

image
image
image

closes #161 
closes #113

### CODE REVIEW

**Review by zhukovdigital — COMMENTED** [review]
@HankC138 It seems like we have a Vercel error, can u check?

**Comment by HankC138** [comment]
> @HankC138 It seems like we have a Vercel error, can u check?

it was a typing error, got that fixed 👍

File: app/components/userAssignment/userLabel.tsx [hunk 1]
```diff
@@ -1,12 +1,153 @@
...
 	const { openModal, closeModal } = useModal();
+	const { setSingleUserPage, userList, setUserList } = useUserDataContext()
+	const { showSummaries, viewer } = useGeneralDataContext()
+	const { projectList, setProjectList, singleProjectPage, setSingleProjectPage } = useProjectsDataContext();
+
 	const isAssignmentProposed = assignment.status === "proposed";
+	const canAssignmentBeDeleted = !assignment.workWeeks.some(
+		(week) => (week.actualHours ?? 0) > 0
+	  );
+	const showArchiveButton = viewer?.id === assignment.assignedUser?.id && !canAssignmentBeDeleted
```
[the remark this answers is not in the document]
Comment (HankC138) [reply app/components/userAssignment/userLabel.tsx:22]: created a separate variable

File: app/components/userAssignment/userLabel.tsx [hunk 2]
```diff
@@ -26,11 +167,10 @@ export const UserLabel = ({ assignment, clickHandler }: UserLabelProps) => {
 			),
 			show: true,
 		},
-		// temporary removal of delete button
-		// {
-		// 	component: () => <button className="text-gray-900 block px-4 py-2 text-sm">Delete</button>,
-		// 	show: true,
-		// },
+		{
+			component: displayDeleteOrArchive().component,
```
[the remark this answers is not in the document]
Comment (HankC138) [reply app/components/userAssignment/userLabel.tsx]: @zhukovdigital I was accidentally blocking the existing hover color by passing a text color, I removed that and the hover is already in place now 👍 .

I also made the changes to the components like you mentioned and no blank space is being rendered now

**Comment by HankC138** [comment]
also made sure to NOT show the ellipsis at all on assignments for users that are not you. But I think that might mean we need to place the ellipsis somewhere else or decide how to adjust the display of labels because they do not line up you can see in the screen shot that the ellipsis menu makes the labels not in line

image

**Comment by HankC138** [comment]
image
would this be better placement? @zhukovdigital

**Comment by zhukovdigital** [comment]
> image would this be better placement? @zhukovdigital

yup, Ithink it will be better. What do uthink about location of the ellipsis menu? Can you raise it a little higher so that it will be aligned  with the name?

**Comment by zhukovdigital** [comment]
> > image would this be better placement? @zhukovdigital
> 
> yup, Ithink it will be better. What do uthink about location of the ellipsis menu? Can you raise it a little higher so that it will be aligned with the name?

Also, what do you think about the sizes of ellipsis menu? 
image
Maybe make it all one size? Like on this screenshot - bigger for more accessibility.

**Comment by HankC138** [comment]
> > > image would this be better placement? @zhukovdigital
> > 
> > 
> > yup, Ithink it will be better. What do uthink about location of the ellipsis menu? Can you raise it a little higher so that it will be aligned with the name?
> 
> Also, what do you think about the sizes of ellipsis menu? image Maybe make it all one size? Like on this screenshot - bigger for more accessibility.

I agree bigger looks better and more clear and the alignment is better as well

**Comment by HankC138** [comment]
@zhukovdigital 
image

@jsonin would like to put this ellipsis on the left hand side in a gutter so everything is aligned. How do you recommend creating a gutter that will keep the alignment of everything regardless if the ellipsis is displayed or not?

**Comment by zhukovdigital** [comment]
> @zhukovdigital image
> 
> @jsonin would like to put this ellipsis on the left hand side in a gutter so everything is aligned. How do you recommend creating a gutter that will keep the alignment of everything regardless if the ellipsis is displayed or not?
It`s depends of the case, here I think u can set fixed width container for ellipsis menu/empty component. So the elements should be aligned. lets try it.

**Comment by HankC138** [comment]
> It`s depends of the case, here I think u can set fixed width container for ellipsis menu/empty component. So the elements should be aligned. lets try it.

image

i think this works well, do you agree? I pushed it if you'd like to review

**Comment by zhukovdigital** [comment]
> > It`s depends of the case, here I think u can set fixed width container for ellipsis menu/empty component. So the elements should be aligned. lets try it.
> 
> image
> i think this works well, do you agree? I pushed it if you'd like to review

yup, looks fine

**Review by zhukovdigital — APPROVED** [review]

**Pull request merged**